project/cmake/CMakeLists.txt

```diff
@@ -460,31 +460,39 @@ configure_file(${PROJECT_SOURCE_DIR}/scripts/common/AddonHelpers.cmake
 configure_file(${PROJECT_SOURCE_DIR}/scripts/common/AddOptions.cmake
                ${CORE_BUILD_DIR}/lib/${APP_NAME_LC}/AddOptions.cmake COPYONLY)
 
+if(VERBOSE)
+  message(STATUS "\n-- INTERNAL VARIABLES ----------------------------------------------------------")
+  message(STATUS "DEPLIBS: ${DEPLIBS}")
+  message(STATUS "core_DEPENDS: ${core_DEPENDS}")
+endif()
+
 # Status
-message(STATUS "#---- CONFIGURATION ----#")
+message(STATUS "\n-- CONFIGURATION ---------------------------------------------------------------")
 foreach(msg ${final_message})
   message(STATUS ${msg})
 endforeach()
 
-if(VERBOSE)
-  message(STATUS "\n#--------------- Internal Variables -------------#")
-  message(STATUS "DEPLIBS: ${DEPLIBS}")
-  message(STATUS "core_DEPENDS: ${core_DEPENDS}")
-  message(STATUS "#---------------------------------------------#")
-  message(STATUS "SYSTEM_LDFLAGS        : ${SYSTEM_LDFLAGS}")
-  message(STATUS "CMAKE_DL_LIBS         : ${CMAKE_DL_LIBS}")
-  message(STATUS "CMAKE_C_FLAGS         : ${CMAKE_C_FLAGS}")
-  message(STATUS "CMAKE_CXX_FLAGS       : ${CMAKE_CXX_FLAGS}")
-  message(STATUS "CMAKE_EXE_LINKER_FLAGS: ${CMAKE_EXE_LINKER_FLAGS}")
-  message(STATUS "#---------------------------------------------#")
-  message(STATUS "bindir     : ${bindir}")
-  message(STATUS "includedir : ${includedir}")
-  message(STATUS "libdir     : ${libdir}")
-  message(STATUS "datarootdir: ${datarootdir}")
-  message(STATUS "#---------------------------------------------#")
-  message(STATUS "GIT_REV: ${APP_SCMID}")
-  message(STATUS "#---------------------------------------------#")
-  message(STATUS "CPACK_GENERATOR       : ${CPACK_GENERATOR}")
-  message(STATUS "CPACK_SOURCE_GENERATOR: ${CPACK_SOURCE_GENERATOR}")
-  message(STATUS "#---------------------------------------------#")
+message(STATUS "\n-- DLOPEN/DLCLOSE LIB ----------------------------------------------------------")
+message(STATUS "  ${CMAKE_DL_LIBS}")
+
+message(STATUS "\n-- FUN WITH FLAGS --------------------------------------------------------------")
+message(STATUS "  CMAKE_C_FLAGS               : ${CMAKE_C_FLAGS}")
+message(STATUS "  CMAKE_CXX_FLAGS             : ${CMAKE_CXX_FLAGS}")
+message(STATUS "  CMAKE_ASM_FLAGS             : ${CMAKE_ASM_FLAGS}")
+message(STATUS "  CMAKE_EXE_LINKER_FLAGS      : ${CMAKE_EXE_LINKER_FLAGS}\n")
+message(STATUS "  SYSTEM_LDFLAGS              : ${SYSTEM_LDFLAGS}\n")
+
+if(NOT WIN32)
+  string(TOUPPER ${CMAKE_BUILD_TYPE} BUILD_TYPE)
```

Comment:
afaik cmake should automatically use the correct ones for the build type merged with external passed flags. If it doesn't then we have to merge them ourselves.